synapse/util/async_helpers.py

```diff
@@ -550,19 +550,21 @@ def failure_cb(val):
     return new_d
 
 
+# This class can't be generic because it uses slots with attrs.
+# See: https://github.com/python-attrs/attrs/issues/313
```

Comment:
Looks like rich worked around this in https://github.com/matrix-org/synapse/commit/d15d241f5ebc978328494ee350d4a0ef8e0070f5

Reply:
NOPE, that's a trap by Rich! I found that on the attrs bug tracker, did the same thing, only to realise that Rich didn't actually get it to work either.

(GitHub does say that it's not part of any branch .... I should have thought more deeply about what that meant :P)

Reply:
Oh, hahaha. I just assumed it was on a PR that got squash merged and then the branch got deleted.

Reply:
:-S